src/Test.sol

```diff
@@ -139,6 +139,256 @@ abstract contract Test is DSTest {
+            emit log            ("Error: a ~= b not satisfied [int]");
+            emit log_named_int  ("  Expected", b);
+            emit log_named_int  ("    Actual", a);
+            emit log_named_int  (" Max Delta", maxDelta);
+            emit log_named_int  ("     Delta", delta);
+            fail();
+        }
+    }
+
+    function assertApproxEq(
+        int256 a,
+        int256 b,
+        int256 maxDelta,
+        string memory err
+    ) internal virtual {
+        int256 delta = a > b ? a - b : b - a;
+
+        if (delta > maxDelta) {
+            emit log_named_string   ("Error", err);
+            emit log_named_int      ("  Expected", b);
+            emit log_named_int      ("    Actual", a);
+            emit log_named_int      (" Max Delta", maxDelta);
+            emit log_named_int      ("     Delta", delta);
+            fail();
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    REL APPROX EQUAL
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function assertRelApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxPercentDelta // An 18 decimal fixed point number, where 1e18 == 100%
+    ) internal virtual {
+        if (b == 0) return assertEq(a, b); // If the expected is 0, actual must be too.
+
+        uint256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log                    ("Error: a ~= b not satisfied [uint]");
+            emit log_named_uint         ("    Expected", b);
+            emit log_named_uint         ("      Actual", a);
+            emit log_named_decimal_uint (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_uint ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
+
+    function assertRelApproxEq(
+        uint256 a,
+        uint256 b,
+        uint256 maxPercentDelta, // An 18 decimal fixed point number, where 1e18 == 100%
+        string memory err
+    ) internal virtual {
+        if (b == 0) return assertEq(a, b); // If the expected is 0, actual must be too.
+
+        uint256 percentDelta = ((a > b ? a - b : b - a) * 1e18) / b;
+
+        if (percentDelta > maxPercentDelta) {
+            emit log_named_string       ("Error", err);
+            emit log_named_uint         ("    Expected", b);
+            emit log_named_uint         ("      Actual", a);
+            emit log_named_decimal_uint (" Max % Delta", maxPercentDelta, 18);
+            emit log_named_decimal_uint ("     % Delta", percentDelta, 18);
+            fail();
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+                                    NOT EQUAL
+    //////////////////////////////////////////////////////////////////////////*/
+
+    function assertNotEq(address a, address b) internal {
```

Comment:
Personally I'd prefer not to include `assertNotEq` methods in forge-std. It adds a lot of boilerplate to have all the same overloads as `assertEq`, and I don't think "not equal" assertions are very useful / good practice. It's always stronger to assert that two values equal, or that a value has changed from `x` to `y`. Asserting things differ is a weak condition to test

Reply:
Okay to drop this, looking through my usage I only have two instances of using notEq and that's for Create2 shenanigans where i could just independently generate the expected address. So agree this is a sign of laziness/code smell

Reply:
Awesome, thanks 👌